### PR TITLE
Favourites context menu

### PR DESCRIPTION
**Pull request by purigarcia**
This PR will fix this issue #786 .
There is a bug. Menu is not refreshed in the gallery neither in the preview of the video when Unfavorite is clicked. @davivel , @masensio could you give me some tips to fix this?
It is pending to change name of variables from keepInSync to favorites. But I's rather wait until the bug is fixed.
Also, we should take into account transifex if we merge that. british english, spanish

### CODE REVIEW

**Comment by davitol** [comment]
:+1:

**Comment by davivel** [comment]
Sorry, but the bug mentioned by @purigarcia is still pening; so this can't be approved by QA.

**Comment by davivel** [comment]
The bug mentioned in the PR is not reproducible. Back to...

File: src/com/owncloud/android/files/FileMenuFilter.java [hunk 1]
```diff
@@ -215,6 +215,20 @@ private void filter(List<Integer> toShow, List <Integer> toHide) {
             toShow.add(R.id.action_send_file);
         }
 
+        // FAVORITES
+        if (mFile == null || downloading || uploading || mFile.isFolder() || mFile.keepInSync()) {
```
Comment (davivel) [inline src/com/owncloud/android/files/FileMenuFilter.java]: Please, rename the method ```OCFile.keepInSync()``` to ```OCFile.isFavorite()```

File: src/com/owncloud/android/files/FileOperationsHelper.java [hunk 2]
```diff
@@ -223,6 +224,24 @@ public void syncFile(OCFile file) {
             mFileActivity.startService(intent);
         }
     }
+
+    public void toggleKeepInSync(OCFile file, boolean isFavorite) {
```
Comment (davivel) [inline src/com/owncloud/android/files/FileOperationsHelper.java]: Please, rename ```toggleKeepInSync(...)``` to ```toggleFavorite```; ```setKeepInSync(...)``` to ```setFavorite(...)``` , etc..

File: src/com/owncloud/android/ui/activity/FileActivity.java [hunk 3]
```diff
@@ -762,6 +768,30 @@ private void onSynchronizeFolderOperationFinish(
...
+                Intent i = new Intent(this, ConflictsResolveActivity.class);
+                i.putExtra(ConflictsResolveActivity.EXTRA_FILE, syncedFile);
+                i.putExtra(ConflictsResolveActivity.EXTRA_ACCOUNT, getAccount());
+                startActivity(i);
+
+            }
+
+        } else {
+            if (operation.transferWasRequested()) {
+
```
Comment (davivel) [inline src/com/owncloud/android/ui/activity/FileActivity.java]: What's the target of this blank block?

I would negate the condition in the ```if``` to apply the code in ```else```directly.

File: src/com/owncloud/android/files/FileMenuFilter.java (hunk 1 above)
Comment (tobiasKaminsky) [inline src/com/owncloud/android/files/FileMenuFilter.java]: Is this a good idea?
As in /core favorite files are something different. And maybe we want to support them in the future (and have them called "isStarred)? Then we would have different namings for same thing.

**Comment by davivel** [comment]
Reviewed, please take a look to my comments.

File: src/com/owncloud/android/files/FileMenuFilter.java (hunk 1 above)
Comment (davivel) [inline src/com/owncloud/android/files/FileMenuFilter.java]: That's something to consider, but right now we have no idea when the integration of the server-side stars will be prioritized (could be soon, could be in two years). On the while I'd prefer the code is as easy to read as possible, a bit of internal coherence will help.

**Comment by davivel** [comment]
:+1:

**Pull request merged**